### PR TITLE
Add `.serve`, `.deploy`, and composite trigger examples to deployment triggers docs

### LINKED ISSUE #11720: Add example of using `flow.deploy` with a trigger
### First check

- [X] I added a descriptive title to this issue.
- [X] I used GitHub search to find a similar request and didn't find it 😇

### Describe the issue

The Automations guide contains an example of `prefect deploy` with a trigger in a YAML file.

### Describe the proposed change

We should add a tab with an example of `flow.deploy` with a trigger.

We could add something like this test.

```
        deployment = await test_flow.to_deployment(
            name="test",
            tags=["price", "luggage"],
            parameters={"name": "Arthur"},
            description="This is a test",
            version="alpha",
            enforce_parameter_schema=True,
            triggers=[
                {
                    "name": "Happiness",
                    "enabled": True,
                    "match": {"prefect.resource.id": "prefect.flow-run.*"},
                    "expect": ["prefect.flow-run.Completed"],
                    "match_related": {
                        "prefect.resource.name": "seed",
                        "prefect.resource.role": "flow",
                    },
                }
            ],
        )
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kevingrismore**
Closes #11720 

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] If this pull request adds new functionality, it includes unit tests that cover the changes
- [x] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [ ] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

File: docs/concepts/automations.md [hunk 1]
```diff
@@ -573,7 +579,101 @@ deployments:
           param_1: "{{ event }}"
 ```
 
-At deployment time, this will create a linked automation that is triggered by events matching your chosen [grammar](/concepts/events/#event-grammar), which will pass the templatable `event` as a parameter to the deployment's flow run.
+This deployment will be run when an `external.resource.pinged` event _and_ an `external.resource.replied` event have been seen from `my.external.resource`:
```
Comment (billpalombi) [inline docs/concepts/automations.md:582]:
```suggestion
This deployment will create a flow run when an `external.resource.pinged` event _and_ an `external.resource.replied` event have been seen from `my.external.resource`:
```

**Review by billpalombi — APPROVED** [review]
One nit, otherwise lgtm

File: docs/concepts/automations.md [hunk 2]
```diff
@@ -556,15 +556,21 @@ Sequence triggers are defined as:
 
 To enable the simple configuration of event-driven deployments, Prefect provides deployment triggers - a shorthand for creating automations that are linked to specific deployments to run them based on the presence or absence of events.
 
+Trigger definitions for deployments are supported in `prefect.yaml`, `.serve`, and `.deploy`. At deployment time, this will create a linked automation that is triggered by events matching your chosen [grammar](/concepts/events/#event-grammar), which will pass the templatable `event` as a parameter to the deployment's flow run.
```
Comment (zzstoatzz) [inline docs/concepts/automations.md:559]:
```suggestion
Trigger definitions for deployments are supported in `prefect.yaml`, `.serve`, and `.deploy`. At deployment time, specified trigger definitions will create a linked automation that is triggered by events matching your chosen [grammar](/concepts/events/#event-grammar). Each trigger definition may include a [jinja template](https://en.wikipedia.org/wiki/Jinja_(template_engine)) to render the triggering `event` as the `parameters` of your  deployment's flow run.
```

File: docs/concepts/automations.md [hunk 3]
```diff
@@ -573,7 +579,101 @@ deployments:
...
+
+For creating deployments with triggers in Python, the trigger types `DeploymentEventTrigger`, `DeploymentMetricTrigger`, `DeploymentCompoundTrigger`, and `DeploymentSequenceTrigger` can be imported from `prefect.events`:
+
+```python
+from prefect import flow
+from prefect.events import DeploymentEventTrigger 
+
+
+@flow(log_prints=True)
+def decorated_fn(param_1):
```
Comment (zzstoatzz) [inline docs/concepts/automations.md:618]:
```suggestion
def decorated_fn(param_1: str):
```

File: docs/concepts/automations.md [hunk 4]
```diff
@@ -573,7 +579,101 @@ deployments:
...
+
+As with prior examples, composite triggers must be supplied with a list of underlying triggers:
+
+```python
+from prefect import flow
+from prefect.events import DeploymentEventTrigger, DeploymentCompoundTrigger
+
+
+@flow(log_prints=True)
+def decorated_fn(param_1):
```
Comment (zzstoatzz) [inline docs/concepts/automations.md:646]:
```suggestion
def decorated_fn(param_1: str):
```

**Review by zzstoatzz — APPROVED** [review]
nice! one nit on wording

**Pull request merged**